default/python/AI/ShipDesignAI.py

```diff
@@ -80,7 +80,7 @@
 
 # Potentially, not adding techs to AIDependencies is intended for testing purposes.
 # Therefore, chat the player only once to inform him about the issue to prevent spam.
-__raised_warnings = []
+_raised_warnings = set()
```

Comment:
Yeah most of the stuff here is based on some extensive work for the fighter branch which was partially relying on a new module. Didn't get to push the most relevant changes to master back then.